pandas-stubs/core/indexes/accessors.pyi

```diff
@@ -434,49 +442,43 @@ class TimedeltaIndexProperties(
     _DatetimeRoundingMethods[TimedeltaIndex],
 ): ...
 
-class _dtDescriptor(CombinedDatetimelikeProperties, Generic[S1]):
+@type_check_only
+class DtDescriptor:
     @overload
-    def __get__(self, instance: Series[Never], owner: Any) -> Never: ...
+    def __get__(self, instance: Series[Never], owner: type[Series]) -> Never: ...
     @overload
     def __get__(
-        self, instance: Series[Timestamp], owner: Any
+        self, instance: Series[Timestamp], owner: type[Series]
     ) -> TimestampProperties: ...
     @overload
     def __get__(
-        self, instance: Series[Timedelta], owner: Any
+        self, instance: Series[Timedelta], owner: type[Series]
     ) -> TimedeltaProperties: ...
+
+@type_check_only
+class ArrayDescriptor:
     @overload
     def __get__(
-        self, instance: Series[S1], owner: Any
-    ) -> CombinedDatetimelikeProperties: ...
-    def round(
-        self,
-        freq: str | BaseOffset | None,
-        ambiguous: Literal["raise", "infer", "NaT"] | bool | np_ndarray_bool = ...,
-        nonexistent: (
-            Literal["shift_forward", "shift_backward", "NaT", "raise"]
-            | timedelta
-            | Timedelta
-        ) = ...,
-    ) -> Series[S1]: ...
-    def floor(
-        self,
-        freq: str | BaseOffset | None,
-        ambiguous: Literal["raise", "infer", "NaT"] | bool | np_ndarray_bool = ...,
-        nonexistent: (
-            Literal["shift_forward", "shift_backward", "NaT", "raise"]
-            | timedelta
-            | Timedelta
-        ) = ...,
-    ) -> Series[S1]: ...
-    def ceil(
-        self,
-        freq: str | BaseOffset | None,
-        ambiguous: Literal["raise", "infer", "NaT"] | bool | np_ndarray_bool = ...,
-        nonexistent: (
-            Literal["shift_forward", "shift_backward", "NaT", "raise"]
-            | timedelta
-            | Timedelta
-        ) = ...,
-    ) -> Series[S1]: ...
-    def as_unit(self, unit: TimeUnit) -> Series[S1]: ...
+        self, instance: IndexOpsMixin[Never], owner: type[IndexOpsMixin]
+    ) -> ExtensionArray: ...
```

Comment:
In contrast to `.dt`, I think keeping this in makes sense.  That's because `.array` works no matter what the underlying dtype of the Series is.